### PR TITLE
BUG: mark success if already at goal

### PR DESCRIPTION
**Pull request by tangkong**
## Description
Simulated Slow motors fail to move in a scan if the first position is the current position. 
Broadly, this issue appears any time sim.slow_motor was moved to its current position with the `wait` argument enabled

## Motivation and Context
This is a functionality that should be included.  
Addresses #945 

## How Has This Been Tested?
Tested interactively, test added to `test_interface.py`

## Where Has This Been Documented?
Code has been commented, related issue, this PR.

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

File: pcdsdevices/sim.py [hunk 1]
```diff
@@ -64,6 +64,12 @@ def _setup_move(self, position, status):
 
         def update_thread(positioner, goal):
             positioner._moving = True
+            if positioner.position == goal:
+                # if already at the requested position, mark success
+                positioner._set_position(goal)  # needed for LiveTable format
```
Comment (tangkong) [inline pcdsdevices/sim.py:69]: I'm not sure why this line is needed.  Without it, LiveTable fails to write new rows after the first.

Comment (klauer) [reply pcdsdevices/sim.py:69]: Does `_set_position` do a subscription callback by chance?

**Comment by tangkong** [comment]
It's possible to fix the issue by removing the `positioner.position != goal` condition from the while loop, but that scares me so I'm going to leave this as is

**Review by klauer — APPROVED** [review]
Looks good to me 👍

**Review by ZLLentz — APPROVED** [review]
Looks good to me, somehow I thought this would be much longer

**Pull request merged**